sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/client.py

```diff
@@ -275,16 +280,21 @@ def unwrap_key(self, algorithm, encrypted_key, **kwargs):
             unwrapped_bytes = result.unwrapped_bytes
 
         """
-
-        result = self._client.unwrap_key(
-            vault_base_url=self._key_id.vault_url,
-            key_name=self._key_id.name,
-            key_version=self._key_id.version,
-            algorithm=algorithm,
-            value=encrypted_key,
-            **kwargs
-        )
-        return UnwrapKeyResult(unwrapped_bytes=result.result)
+        local_key = self._get_local_key()
+        if local_key and local_key.is_private_key():
+            if "unwrapKey" not in self._allowed_ops:
+                raise AzureError("This client doesn't have 'keys/unwrapKey' permission")
+            result = local_key.unwrap_key(encrypted_key, **kwargs)
+        else:
+            result = self._client.unwrap_key(
+                vault_base_url=self._key_id.vault_url,
```

Comment:
is there case where the key is a local_key but is_private_key is false, an _key_id is not set? does this code need to guard against that?

Reply:
No, init will raise when `_key_id` isn't set.